### PR TITLE
fix NaNs in AutoML: regression

### PR DESCRIPTION
**Pull request by LukaszCmielowski**
### Summary                                                                                                                                                                                                                                                 
  Implements consistent data cleansing across AutoML components to handle infinite values, duplicates, and missing labels before model training, preventing AutoGluon failures.
                                                                                                                                                                                                                                                              
### Problem                                               
  AutoGluon rejects datasets with infinite values or NaN labels, causing cryptic training failures. Different sampling methods also handled missing labels inconsistently.                                                                                    
                                                            
### Solution
  Apply uniform cleansing in both data loader and training components:
  1. Replace `±infinity` with `NaN`
  2. Validate label column exists                                                                                                                                                                                                                             
  3. Drop full-row duplicates                                                                                                                                                                                                                                 
  4. Drop rows with missing labels                                                                                                                                                                                                                            
                                                                                                                                                                                                                                                              
  This prevents training failures and ensures data quality consistency across the pipeline.                                                                                                                                                                   
   
  ### Key Changes                                                                                                                                                                                                                                             
                                                            
  **Data Loader Component:**                                                                                                                                                                                                                                  
  - Cleansing after sampling, before splitting
  - Improved error messages with better validation order                                                                                                                                                                                                      
  - Consistent behavior across all sampling methods (stratified, random, first_n_rows)                                                                                                                                                                        
                                                                                                                                                                                                                                                              
  **Training Component:**                                                                                                                                                                                                                                     
  - `_clean_frame_after_read()` helper applies identical cleansing to train/test/extra-train data
  - Gracefully handles empty extra-train data                                                                                                                                                                                                                 
  - Added duplicate-drop logging for consistency                                                                                                                                                                                                              
   
  **Tests & Code Quality:**                                                                                                                                                                                                                                   
  - 16 new unit tests covering edge cases (all-infinity, all-missing, empty datasets)
  - Realistic test data: `taxi_trip_pricing_sample.csv`
  - Fixed `PipelineConfig` pytest warning (was `TestConfig`)                                                                                                                                                                                                  
  - Fixed string normalization in config loader
                                                                                                                                                                                                                                                              ### Breaking Changes                                      
  None. Cleansing is transparent and only removes rows that would cause training failures.

### Pre-Submission Checklist

- [x] All tests and CI checks pass
- [x] Pre-commit hooks pass without errors
- [x] You have [signed off your commits](https://www.kubeflow.org/docs/about/contributing/#sign-off-your-commits)
- [x] The title for your pull request (PR) should follow our title convention.
  [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### Additional Checklist Items for New or Updated Components/Pipelines

- [x] `metadata.yaml` includes fresh `lastVerified` timestamp
- [x] All [required files](https://github.com/kubeflow/pipelines-components/blob/main/docs/CONTRIBUTING.md#required-files)
  are present and complete
- [x] OWNERS file lists appropriate maintainers
- [x] README provides clear documentation with usage examples
- [x] Component follows `snake_case` naming convention
- [x] No security vulnerabilities in dependencies
- [x] Containerfile included if using a custom base image

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

## Release Notes

* **Bug Fixes**
  * Improved data preprocessing to handle edge cases: infinite values are now converted to missing values, duplicate rows are removed, and rows with missing labels are dropped before training.
  * Enhanced validation to reject invalid datasets and provide clearer error messages.

* **Documentation**
  * Updated preprocessing flow documentation to clarify data cleaning steps and validation requirements.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

### CODE REVIEW

**Review by DorotaDR — APPROVED** [review]
/lgtm

File: components/data_processing/automl/tabular_data_loader/component.py [hunk 1]
```diff
@@ -305,6 +315,43 @@ def load_data_in_batches(
         label_column=label_column,
     )
 
+    sampled_dataframe.replace([math.inf, -math.inf], float("nan"), inplace=True)
+
+    if label_column not in sampled_dataframe.columns:
+        raise ValueError(
+            f"Label column {label_column!r} not found in the dataset. "
+            f"Available columns: {list(sampled_dataframe.columns)}"
+        )
```
Comment (Mateusz-Switala) [inline components/data_processing/automl/tabular_data_loader/component.py]:
```suggestion
    if label_column not in sampled_dataframe.columns:
        raise ValueError(
            f"Label column {label_column!r} not found in the dataset. "
            f"Available columns: {list(sampled_dataframe.columns)}"
        )
        
    sampled_dataframe.replace([math.inf, -math.inf], float("nan"), inplace=True)

    
```
The label column check is a structural validation that doesn't depend on data values. It should happen before any data manipulation.

File: components/training/automl/autogluon_models_training/component.py [hunk 2]
```diff
@@ -117,11 +123,63 @@ def autogluon_models_training(
...
+                n_drop,
+                role,
+                label_column,
+                len(df),
+            )
+
     # 1. models selection stage
 
     train_data_df = pd.read_csv(train_data_path)
+    _clean_frame_after_read(train_data_df, role="train")
```
Comment (Mateusz-Switala) [inline components/training/automl/autogluon_models_training/component.py]: Aren't we doing the same thing in Data Loader? After all, it's the same data.

File: pipelines/training/automl/autogluon_tabular_training_pipeline/tests/test_tabular_config_loader.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
+"""Tests for tabular integration test config JSON loading."""
```
Comment (Mateusz-Switala) [inline pipelines/training/automl/autogluon_tabular_training_pipeline/tests/test_tabular_config_loader.py:1]: Why do we need these tests? I mean, isn't this just testing the tests? Test configs is part of integration tests.

Comment (LukaszCmielowski) [reply pipelines/training/automl/autogluon_tabular_training_pipeline/tests/test_tabular_config_loader.py:1]: I'd keep it - there is not harm and we just make sure the config is correct.

**Review by Mateusz-Switala — APPROVED** [review]
lgtm
